test/Microsoft.Azure.WebJobs.Host.EndToEndTests/Scale/ScaleHostEndToEndTests.cs

```diff
@@ -86,28 +93,38 @@ public async Task ScaleManager_GetScaleStatusAsync_ReturnsExpected(bool tbsEnabl
                 services.AddSingleton<IConcurrencyStatusRepository, TestConcurrencyStatusRepository>();
             })
             .ConfigureWebJobsScale((context, builder) =>
-                {
-                    builder.UseHostId(hostId);
-                },
-                scaleOptions =>
-                {
-                    scaleOptions.IsTargetScalingEnabled = tbsEnabled;
-                    scaleOptions.MetricsPurgeEnabled = false;
-                    scaleOptions.ScaleMetricsMaxAge = TimeSpan.FromMinutes(4);
-                    scaleOptions.IsRuntimeScalingEnabled = true;
-                });
-
-            foreach (TriggerMetadata t in triggerMetadata)
             {
-                if (t.Type == ATriggerType)
-                {
-                    hostBuilder.ConfigureTestExtensionAScale(t);
-                }
-                else
+                builder.UseHostId(hostId);
+
+                // Get the interface type you want to find
+                Type interfaceType = typeof(IWebJobsStartup2);
```

Comment:
Would be good to clean this code up into a pattern we can point people to to follow for actual ScaleHost applications. I think you can simply use the existing [UseExternalStartup](https://github.com/Azure/azure-webjobs-sdk/blob/8960a7294b854b65c2b416b23dfc2fcc80574725/src/Microsoft.Azure.WebJobs.Host/Hosting/WebJobsBuilderExtensions.cs#L206) method, passing in your own discoverer and correct context type? That way we leverage all the existing activation logic.

Reply:
Yesh using [UseExternalStartup](https://github.com/Azure/azure-webjobs-sdk/blob/8960a7294b854b65c2b416b23dfc2fcc80574725/src/Microsoft.Azure.WebJobs.Host/Hosting/WebJobsBuilderExtensions.cs#L206) makes sense.